### ISSUE TITLE
Providing Type of parameterized collections in JSON

### ISSUE THREAD

**Issue by johanhaleby** [jayway/rest-assured#295]
_From [fcamb...@gmail.com](https://code.google.com/u/111948726400665388556/) on January 10, 2012 09:08:45_

What steps will reproduce the problem? 1. Provide a REST service returning a typed JSON array like this one : [ { "foo": "foo", "bar": "bar" }, { "foo": "hello", "bar": "world" } ]
2. Try to call this service in rest-assured, then map it on a List<Foo> (where Foo contains foo and bar fields) :
List<Foo> myList = expect().statusCode(200).when().get("/foos").as(List.class);
3. Since we have reified generics in Java, rest-assured cannot predict the parameterized type of the List and then, will map the return type in a List<Map> instead of a List<Foo>

Idea would be to provide a GSON-like approach with something looking like TypeToken.

Discussion started on google group here : https://groups.google.com/forum/#!topic/rest-assured/0GIQQfzC_6c

_Original issue: http://code.google.com/p/rest-assured/issues/detail?id=141_

**Comment by johanhaleby** [comment]
_From [johan.ha...@gmail.com](https://code.google.com/u/105676376875942041029/) on January 10, 2012 02:26:40_

**Status:** Accepted  
  **Labels:** -Type-Defect Type-Enhancement

**Comment by johanhaleby** [comment]
_From [johan.ha...@gmail.com](https://code.google.com/u/105676376875942041029/) on February 01, 2012 23:18:08_

**Labels:** 1.7

**Comment by chrislong** [comment]
This seems to have been addressed. I found a solution on the user group: 
https://groups.google.com/d/msg/rest-assured/a0m6wKzzy6c/OdP6_GKaCQAJ

**Comment by johanhaleby** [comment]
Yes it works for shallow data structures such as List<String> but it doesn't work for deeper onces such as Map<String, Map<String, String>>

**Comment by chrislong** [comment]
Ah, I understand. Thanks. FWIW, I like the TypeToken idea.
